src/load_balancer/load_balancer.rs

```diff
@@ -35,14 +35,15 @@ impl LoadBalancer {
 
         // Create a new URI from the worker URI
         let new_uri = Uri::from_str(&worker_uri).map_err(LoadBalancerError::InvalidUri)?;
+        let new_address = format!("{}:{}", new_uri.host().unwrap(), new_uri.port().unwrap());
```

Comment:
Please consider managing errors in a next iteration; unwrap is ok for initial exploration, but not ok for finalized code.